### PR TITLE
feat: Support ProxyConfig settings to OAuth2 configuration for Alertmanager

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

The next PR for #6735 

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

```
Alertmanager would need to be covered too (or it's deferred to a follow-up PR but in this case, it should be noted in the documentation)
```
Originally posted by @simonpasquier  in https://github.com/prometheus-operator/prometheus-operator/pull/6735#pullrequestreview-2224354618

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Support ProxyConfig settings to OAuth2 configuration for Alertmanager
```

### CODE REVIEW

**Comment by dongjiang1989** [comment]
cc @simonpasquier  PTAL

File: pkg/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -87,19 +87,19 @@ type ProxyConfig struct {
 	// that should be excluded from proxying. IP and domain names can
 	// contain port numbers.
 	//
-	// It requires Prometheus >= v2.43.0.
+	// It requires Prometheus >= v2.43.0 or Alertmanager >= 0.25.0.
```
Comment (dongjiang1989) [inline pkg/apis/monitoring/v1/types.go:90]: `ProxyConfig` is reused by Prometheus and Alertmanager, so this description is changed.
/cc @simonpasquier PTAL

**Review by simonpasquier — COMMENTED** [review]
Thanks for the PR! I realized that #6818 was incomplete so I submitted https://github.com/prometheus-operator/prometheus-operator/pull/6861 which we would need to merge before going further with this PR. 
I'd also recommend to split into several PRs:
* 1 for adding proxy support in OAuth2 config.
* 1 for supporting new proxy fields into the AlertmanagerConfig CRD for the HTTP clients.
* 1 for supporting new proxy fields  into the AlertmanagerConfig CRD for the OAuth2 clients.

**Comment by dongjiang1989** [comment]
- [ ] adding proxy support in OAuth2 config.
- [ ] supporting new proxy fields into the AlertmanagerConfig CRD for the HTTP clients. #6864 
- [ ] supporting new proxy fields into the AlertmanagerConfig CRD for the OAuth2 clients.

**Comment by Tanc010** [comment]
> TODO list:
> 
> * [ ]  adding proxy support in OAuth2 config.
> * [x]  supporting new proxy fields into the AlertmanagerConfig CRD for the HTTP clients. [feat: Support ProxyConfig fields into the AlertmanagerConfig CRD for the HTTP clients #6864](https://github.com/prometheus-operator/prometheus-operator/pull/6864)
> * [ ]  supporting new proxy fields into the AlertmanagerConfig CRD for the OAuth2 clients.

Is the first task PR [#6735](feat: Support OAuth2 TLSConfig and ProxyConfig settings)[feat: Support OAuth2 TLSConfig and ProxyConfig settings](https://github.com/prometheus-operator/prometheus-operator/pull/6735/files#top)?

**Comment by dongjiang1989** [comment]
> TODO list:
> 
> * [ ]  adding proxy support in OAuth2 config.
> * [x]  supporting new proxy fields into the AlertmanagerConfig CRD for the HTTP clients. [feat: Support ProxyConfig fields into the AlertmanagerConfig CRD for the HTTP clients #6864](https://github.com/prometheus-operator/prometheus-operator/pull/6864)
> * [x]  supporting new proxy fields into the AlertmanagerConfig CRD for the OAuth2 clients. [supporting new proxy fields into the AlertmanagerConfig CRD for the O… #6922](https://github.com/prometheus-operator/prometheus-operator/pull/6922)

All TODO list done. Close this PR.

**Pull request closed without merging**